### PR TITLE
Refactor: using Global state in Board.jsx

### PR DESCRIPTION
**Pull request by JonathanYesh279**

### CODE REVIEW

File: src/services/board.service.local.js [hunk 1]
```diff
@@ -310,23 +310,13 @@ async function remove(boardId) {
...
-
   if (board._id) {
-    const boardToUpdate = {
-      _id: board._id,
-      title: board.title,
-    };
-
-    savedBoard = await storageService.put(STORAGE_KEY, boardToUpdate);
+    const updatedBoard = await storageService.put(STORAGE_KEY, board);
+    return updatedBoard;
```
Comment (tomercarolla) [inline src/services/board.service.local.js:315]: you dont need return here, you can return await storageServ....

File: src/components/board/Board.jsx [hunk 2]
```diff
@@ -245,17 +269,18 @@ export function Board() {
...
-            Object.values(board.orderedGroupsIds).map((groupId) => {
+            board.orderedGroupsIds.map((groupId) => {
               const group = board.groups[groupId];
               const tasks = group.tasksIds.map((taskId) => board.tasks[taskId]);
-
+              
               return (
                 <Column
-                  key={group.id}
+                  key={`${group.id}-${group.tasksIds.join('-')}`} // update the key prop to include tasks Ids, it forces the component to re-render when tasks are updated
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:283]: I dont think add task id, affects the task render

File: src/components/board/Board.jsx [hunk 3]
```diff
@@ -245,17 +269,18 @@ export function Board() {
 
   return (
     <div className="board">
+      {console.log('Rendering board with data:', board)}
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:272]: remove console log

File: src/components/board/Board.jsx [hunk 4]
```diff
@@ -6,42 +6,64 @@ import { useCallback, useEffect, useRef } from 'react';
...
-      console.log('moveTask newStartGroupData ', newStartGroupData);
-      console.log('moveTask newFinishGroupData ', newFinishGroupData);
+ const moveTask = useCallback(
+   ({
+     taskIndexInStartGroup,
+     sourceGroupId,
+     destinationGroupId,
+     itemIndexInFinishGroup,
+   }) => {
+     if (!board) return;
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:32]: why you need here board condition?

File: src/components/board/Board.jsx [hunk 5]
```diff
@@ -56,67 +78,69 @@ export function Board() {
...
-    [board],
-  );
+  //   },
+  //   [board],
+  // );
 
   const reorderTask = useCallback(
     ({ groupId, startIndex, finishIndex }) => {
-      const sourceGroupData = board.groups[groupId];
+      if (!board) return;
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:87]: why you need here board condition?

File: src/components/board/Board.jsx [hunk 6]
```diff
@@ -6,42 +6,64 @@ import { useCallback, useEffect, useRef } from 'react';
...
+ const moveTask = useCallback(
+   ({
+     taskIndexInStartGroup,
+     sourceGroupId,
+     destinationGroupId,
+     itemIndexInFinishGroup,
+   }) => {
+     if (!board) return;
+
+     const updatedBoard = JSON.parse(JSON.stringify(board)); // Deep clone
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:34]: why???

File: src/components/board/Board.jsx [hunk 7]
```diff
@@ -6,42 +6,64 @@ import { useCallback, useEffect, useRef } from 'react';
...
 import { useDraggable } from 'react-use-draggable-scroll';
-import { loadBoard } from '../../store/board/board.actions.js';
+import { loadBoard, updateBoard } from '../../store/board/board.actions.js';
 import { BoardHeader } from './components/BoardHeader.jsx';
 import { Column } from './components/Column.jsx';
 
 export function Board() {
   const { boardId } = useParams();
   const board = useSelector((state) => state.boardModule.board);
+  console.log('Board data:', board);
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:16]: remove console log

File: src/components/board/Board.jsx [hunk 8]
```diff
@@ -56,67 +78,69 @@ export function Board() {
...
+  //   [board],
+  // );
 
   const reorderGroup = useCallback(
     ({ startIndex, finishIndex }) => {
-      const updatedGroups = reorder({
-        list: board.orderedGroupsIds,
+      if (!board) return;
+
+      const updatedBoard = JSON.parse(JSON.stringify(board)); // Deep clone
```
Comment (tomercarolla) [inline src/components/board/Board.jsx:117]: same

File: src/components/board/Board.jsx (hunk 6 above)
Comment (JonathanYesh279) [reply src/components/board/Board.jsx:34]: when you change nested states it is better to use deep clone (groups, tasks..) . this was the issue when we tried to render the changes after dnd actions, and didn't see any change. shallow clone like ...board will not work beacuse we need to create new objects for these nested states/properties..
i am not sure this is the optimised way to make a deep clone but we can ask yonatan about it.

File: src/components/board/Board.jsx (hunk 4 above)
Comment (JonathanYesh279) [reply src/components/board/Board.jsx:32]: We don't have to, but i used to do it every where i am afraid to get null/undefined. but acutally we can make it only where we fetch board and not inside the group/task... i will remove it if u dont want...

**Review by tomercarolla — APPROVED** [review]

**Pull request merged**